src/fileprovider.cs

```diff
@@ -0,0 +1,301 @@
+	}
+
+	[iOS (11,0)]
+	[DisableDefaultCtor]
+	[BaseType (typeof (NSObject))]
+	interface NSFileProviderDomain {
+
+		[Export ("initWithIdentifier:displayName:pathRelativeToDocumentStorage:")]
+		IntPtr Constructor (string identifier, string displayName, string pathRelativeToDocumentStorage);
+
+		[Export ("identifier")]
+		string Identifier { get; }
+
+		[Export ("displayName")]
+		string DisplayName { get; }
+
+		[Export ("pathRelativeToDocumentStorage")]
+		string PathRelativeToDocumentStorage { get; }
+	}
+
+	interface INSFileProviderEnumerationObserver { }
+
+	[iOS (11,0)]
+	[Protocol]
+	interface NSFileProviderEnumerationObserver {
+
+		[Abstract]
+		[Export ("didEnumerateItems:")]
+		void DidEnumerateItems (INSFileProviderItem [] updatedItems);
+
+		[Abstract]
+		[Export ("finishEnumeratingUpToPage:")]
+		void FinishEnumerating ([NullAllowed] NSData nextPage);
+
+		[Abstract]
+		[Export ("finishEnumeratingWithError:")]
+		void FinishEnumerating (NSError error);
+	}
+
+	interface INSFileProviderChangeObserver { }
+
+	[iOS (11,0)]
+	[Protocol]
+	interface NSFileProviderChangeObserver {
+
+		[Abstract]
+		[Export ("didUpdateItems:")]
+		void DidUpdateItems (INSFileProviderItem [] updatedItems);
+
+		[Abstract]
+		[Export ("didDeleteItemsWithIdentifiers:")]
+		void DidDeleteItems (string [] deletedItemIdentifiers);
+
+		[Abstract]
+		[Export ("finishEnumeratingChangesUpToSyncAnchor:moreComing:")]
+		void FinishEnumeratingChanges (NSData anchor, bool moreComing);
+
+		[Abstract]
+		[Export ("finishEnumeratingWithError:")]
+		void FinishEnumerating (NSError error);
+	}
+
+	interface INSFileProviderEnumerator { }
+
+	[iOS (11,0)]
+	[Protocol]
+	interface NSFileProviderEnumerator {
+
+		[Abstract]
+		[Export ("invalidate")]
+		void Invalidate ();
+
+		[Abstract]
+		[Export ("enumerateItemsForObserver:startingAtPage:")]
+		void EnumerateItems (INSFileProviderEnumerationObserver observer, NSData page);
+
+		[Export ("enumerateChangesForObserver:fromSyncAnchor:")]
+		void EnumerateChanges (INSFileProviderChangeObserver observer, NSData anchor);
+
+		[Export ("currentSyncAnchorWithCompletionHandler:")]
+		void CurrentSyncAnchor (Action<NSData> completionHandler);
+	}
+
+	interface INSFileProviderItem { }
+
+	[iOS (11,0)]
+	[Protocol]
+	interface NSFileProviderItem {
+
+		[Advice ("Use 'NSFileProviderItemIdentifierExtensions.GetValue (ItemIdentifier)' to get a 'NSFileProviderItemIdentifier' enum.")]
+		[Abstract]
+		[Export ("itemIdentifier")]
+		NSString ItemIdentifier { get; }
+
+		[Advice ("Use 'NSFileProviderItemIdentifierExtensions.GetValue (ParentItemIdentifier)' to get a 'NSFileProviderItemIdentifier' enum.")]
+		[Abstract]
+		[Export ("parentItemIdentifier")]
+		NSString ParentItemIdentifier { get; }
+
+		[Abstract]
+		[Export ("filename")]
+		string Filename { get; }
+
+		[Abstract]
+		[Export ("typeIdentifier")]
+		string TypeIdentifier { get; }
+
+		[Export ("capabilities")]
+		NSFileProviderItemCapabilities GetCapabilities ();
```

Comment:
You're right, I didn't notice that the properties were required and the methods optional.